Moshiur

moshiur9185 opened the pull request.

mujahidinfo committed:
update code

moshiur9185 committed:
Added new forms for lesson and parent management, updated validation schemas, and improved error handling in actions and  add data in database dymamicly.

moshiur9185 committed:
Merge pull request #1 from Genres-Agency/moshiur

 Added new forms for lesson and parent management, updated validation…

AttoDeba1 commented on src/app/(dashboard)/layout.tsx:
```diff
@@ -17,7 +17,7 @@ export default function DashboardLayout({
         >
           <Image src="/logo.png" alt="logo" width={32} height={32} />
-          <span className="hidden lg:block font-bold">SchooLama</span>
+          <span className="hidden lg:block font-bold">GenSchool</span>
```
Why did you change the name? 
It might be better to make the school’s name configurable through an environment variable.

AttoDeba1 replied:
To take it a step further, we could explore integrating a platform customization system to handle such changes dynamically.

AttoDeba1 commented on src/components/forms/LessonForm.tsx:
```diff
@@ -0,0 +1,137 @@
+    //   return;
+    // }
+    formAction(data);
+    console.log("sdjkjs====",data);
```
```suggestion
```

AttoDeba1 commented on src/components/forms/TeacherForm.tsx:
```diff
@@ -144,7 +141,7 @@ const TeacherForm = ({
             register={register}
             error={errors?.id}
-            hidden
+            // hidden
```
Why?

AttoDeba1 commented on src/lib/actions.ts:
```diff
@@ -466,3 +522,92 @@ export const deleteExam = async (
+
+    await prisma.parent.delete({
+      where: {
+        id: id,
```
```suggestion
        id
```

moshiur9185 committed:
Update

moshiur9185 committed:
Update work

moshiur9185 committed:
Update  work and Fixed some issue

moshiur9185 committed:
Merge pull request #2 from Genres-Agency/moshiur

Moshiur

moshiur9185 committed:
Update some feature

moshiur9185 committed:
Refactor FinanceChart to use BarChart instead of LineChart for better data visualization

moshiur9185 committed:
Add upload functionality and refactor related components

- Implement upload routes and handlers in the API
- Create UploadButton and UploadDropzone components
- Update Tailwind configuration to include new paths
- Refactor schema to replace parentId with parentNId and parentName
- Update sign-in page for user authentication

moshiur9185 committed:
Refactor Announcement model and related components to use startDate and endDate instead of date; update form validation and API routes accordingly

moshiur9185 committed:
Remove parent-related components and queries from dashboard pages; adjust layout widths for consistency

moshiur9185 committed:
Refactor dashboard layout, add admission page, and update teacher and student form with image upload functionality

moshiur9185 committed:
Refactor image upload functionality in forms; update teacher and student pages for improved layout and image handling

moshiur9185 committed:
Add select-none class to Announcements component for improved text selection behavior

moshiur9185 committed:
Add admission management features, including form, PDF download, and database schema

moshiur9185 committed:
feat: Add form validation schemas, CRUD operations for teachers/students/parents, and menu components. Include admission form schema, result management.

moshiur9185 committed:
Add CRUD operations for results, and announcements. Include admission management and image upload functionality

moshiur9185 committed:
feat: Add new UI components including Skeleton, Tooltip, Separator, and Input; refactor Dashboard layout and Navbar for improved structure

moshiur9185 committed:
refactor: Update Prisma schema for improved clarity and consistency; enhance loading component styles and menu interactions; adjust form validation for student ID

moshiur9185 committed:
feat: Update ResultForm component and enhance result validation schema; refactor admin seeding logic and adjust database schema for results

moshiur9185 committed:
refactor: Update ResultList structure to include student and subject details; adjust data fetching and rendering logic

moshiur9185 committed:
refactor: Update subjectId type in Result model and adjust related logic; enhance form validation and data handling for results

moshiur9185 committed:
feat: Implement attendance tracking with new Attendance model; enhance attendance form validation and create AttendanceList page

moshiur9185 committed:
refactor: Remove studentId and lessonId from Attendance model; update AttendanceForm and actions for improved data handling and validation

moshiur9185 committed:
refactor: Update delete action mapping for announcements and attendance; remove console logs from AttendanceForm and actions for cleaner code

moshiur9185 committed:
refactor: Enhance subject validation and update form handling; replace ResultForm2 with ResultForm for improved structure

moshiur9185 committed:
refactor: Update table alignment and visibility settings; enhance user interface consistency across components

moshiur9185 committed:
refactor: Replace lessonId with classId in exam-related components and schemas; update form handling for improved clarity and data integrity

moshiur9185 committed:
refactor: Remove gradeId from Class model and related components; update ClassForm and validation schema for improved data handling

moshiur9185 committed:
Merge pull request #3 from Genres-Agency/moshiur

Moshiur

moshiur9185 committed:
refactor: Improve table cell alignment and visibility in attendance, results, and subjects pages for enhanced UI consistency

moshiur9185 committed:
Merge pull request #4 from Genres-Agency/moshiur

refactor: Improve table cell alignment and visibility in attendance, …

moshiur9185 committed:
refactor: Update form imports to use specific action files for improved modularity and maintainability

moshiur9185 committed:
refactor: Update student relation in Result model to use username; adjust related components for improved data consistency and access control

moshiur9185 committed:
feat: Add theme support with ThemeProvider and ThemeSwitcher; enhance UI with dark mode styling across components

moshiur9185 committed:
refactor: update dynamic attendence chart, Enhance dark mode support and UI consistency across components; adjust layouts and styles for improved user experience

moshiur9185 committed:
feat: Add Finance model and related forms; enhance UI with dark mode support across various forms

moshiur9185 committed:
feat: Add Payment model and form; implement manual payment processing functionality and UI integration

moshiur9185 committed:
feat: Add student by PaymentList component; update forms to handle submission state and improve UI consistency

moshiur9185 committed:
feat: add pdf download forom payment history, Add fees field to Class model; update related forms and components for improved payment handling and pagination

moshiur9185 committed:
feat: integrate Radix UI collapsible component; enhance payment PDF download functionality and update UI elements for improved clarity

moshiur9185 committed:
feat: update form components to handle date formatting and improve searchParams handling; create exam route update funtionality

moshiur9185 committed:
feat: fixed dome issue, update authentication calls to use async/await pattern; modify exam routine

moshiur9185 committed:
feat: add Admit Card component and PDF download functionality; update exam access permissions and remove deprecated exam schema

moshiur9185 committed:
feat: update dependencies; add Emergency Numbers component, enhance Admit Card button with additional student details

moshiur9185 committed:
feat: update student identifier from ID to username across payment and admit card components; enhance Admit Card PDF with student image and signature section

moshiur9185 committed:
feat: integrate FinanceChartContainer for dynamic finance data; update AdminPage to use FinanceChartContainer; clean up console logs and improve table formatting

moshiur9185 committed:
Updating

Op-Pautu commented:
why u tryna make a pr here 💀

moshiur9185 committed:
Updated task

moshiur9185 committed:
Update some work

moshiur9185 committed:
code transfar with ataullah

moshiur9185 committed:
Update some work

ataullah1 committed:
Implement authentication provider and enhance error handling in auth pages and fix all authentication and dashboard redirect issue resolved

moshiur9185 committed:
Fixed authentication issue update role based authentication, update folder structure and  include nextjs-toploader

moshiur9185 committed:
Update authorization

moshiur9185 committed:
Update users action and admin autherization

moshiur9185 committed:
Fixed student rolebased authorization

moshiur9185 committed:
add online payment option

moshiur9185 committed:
Update Teacher authorization functionality

moshiur9185 committed:
Update UI and some bugs fixed

moshiur9185 committed:
Fixed issue

moshiur9185 committed:
Update some task

moshiur9185 committed:
Update some task

moshiur9185 committed:
Fixed some issue and update some task

moshiur9185 committed:
Fixed finance data dynamic shows in finance chart

moshiur9185 committed:
Update dashboard menu bar and fixed some issue

moshiur9185 committed:
Fixed some bugs

moshiur9185 committed:
Fixed some bugs

moshiur9185 committed:
Fixed some bugs

moshiur9185 committed:
Fixed some bugs

moshiur9185 committed:
Modify side scroll bar

moshiur9185 committed:
Add class material system and update some UI

moshiur9185 committed:
Fixed some bugs issue

moshiur9185 committed:
Update uploadthing darag and drop option and fixed some bugs and update some UI

moshiur9185 committed:
Fixed ruslt create bugs

moshiur9185 committed:
Update some UI and Task

moshiur9185 committed:
Fixed build issue

moshiur9185 committed:
Update database migrations

moshiur9185 committed:
Fixed class materiral data create issue

moshiur9185 committed:
Update mobile responsive  in admin dashboard

moshiur9185 committed:
Update home page navigation user profile button

moshiur9185 committed:
Fixed modal position

moshiur9185 committed:
Fixed sort functionality and fixed some ui

moshiur9185 committed:
Implement sorting all list page

moshiur9185 committed:
Update some ui and optimize some code

moshiur9185 committed:
Update Ui

moshiur9185 committed:
Fixed student list table header

moshiur9185 committed:
Update form UI

moshiur9185 committed:
Fixed all form UI issoue

moshiur9185 committed:
Fixed new amnouncment coponent import issue

moshiur9185 committed:
Update some UI and fixed finance chart ui issue and add no records available option

moshiur9185 committed:
Fixed some mobile responsive UI

moshiur9185 committed:
Fixed student and teacher login bugs

moshiur9185 committed:
Update scalaton UI
